### PR TITLE
somes fixes + docker build optimisation + cli verbose option

### PR DESCRIPTION
**Pull request by mathyba**
Some fixes made when checking with and without docker

### CODE REVIEW

File: src/slot_checker.py [hunk 1]
```diff
@@ -219,16 +225,20 @@ def quit(self):
         default='config.yml',
         help='config file'
     )
+    parser.add_argument('-v', '--verbose', action='count', default=0, help='include debugging logs')
```
Comment (nirae) [inline src/slot_checker.py]: maybe use the action 'store_true'? for a boolean argument, it can be simpler

File: src/slot_checker.py [hunk 2]
```diff
@@ -72,6 +77,7 @@ def wrapper(*args, **kwargs):
     @check_signin
     def get_project_slots(self, project, start, end):
         if project == 'debug_my_slots':
+        if DEBUG_LEVEL == log.DEBUG:
```
Comment (nirae) [inline src/slot_checker.py]: Maybe a merge error:

File "/usr/src/app/./slot_checker.py", line 80
if DEBUG_LEVEL == log.DEBUG:
IndentationError: expected an indented block

The correct line is:
if project == 'debug_my_slots':

**Review by nirae — CHANGES_REQUESTED** [review]
Thanks!

 - The version of the docker-compose (3) seems to be not compatible with the volumes long syntax. The command 'make up' produce the error:
 
```
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.slot_checker.volumes contains an invalid type, it should be a string
make: *** [Makefile:7 : up] Erreur 1
```

Fixed if upgrade the docker-compose file version to '3.7'

- In the Makefile, the docker-compose up need the option -d to run in detached mode

- Why 'touch config.yml' in the Dockerfile?

**Comment by nirae** [comment]
I added a build CI in master so I merged master on your PR so that you can get the CI

File: src/slot_checker.py (hunk 1 above)
Comment (mathyba) [reply src/slot_checker.py]: you are right, I didn't check the doc carefully enough :)

**Comment by mathyba** [comment]
On why using bind mount rather than volumes:
With previous use of docker-compose volumes, if source or target doesn't exist, a directory is created.
This causes crashes and is confusing for the user.

Straight from the [doc](https://docs.docker.com/storage/bind-mounts/):
> If you use -v or --volume to bind-mount a file or directory that does not yet exist on the Docker host, -v creates the endpoint for you. It is always created as a directory.
> If you use --mount to bind-mount a file or directory that does not yet exist on the Docker host, Docker does not automatically create it for you, but generates an error.
The latter is actually the desired behaviour: if the user fails to create config.yml, there'll be an error

File: src/slot_checker.py (hunk 2 above)
Comment (mathyba) [reply src/slot_checker.py]: definitely a rebase error

**Comment by nirae** [comment]
Ok so using the bind mount for the volume in the compose file is a good idea!

**Review by nirae — APPROVED** [review]

**Comment by nirae** [comment]
Always a pleasure to work with you @mathyba

**Pull request merged**